### PR TITLE
Sync unavailableNodeCount at policy Reconcile

### PR DESCRIPTION
**Pull request by rhrazdil**
Add syncPolicyMaxUnavailableCount method that checks number of
Progressing enactments matches the policy unavailableNodeCount.

If counts don't match, it's caused by a handler being restarted
before decrementing the unavailableNodeCount. In that case, set
the policy unavailableNodeCount to the number of currently progressing
enactments.

Signed-off-by: Radim Hrazdil <rhrazdil@redhat.com>

**Is this a BUG FIX or a FEATURE ?**:

> Uncomment only one, leave it on its own line:
>
/kind bug
> /kind enhancement

**What this PR does / why we need it**:

**Special notes for your reviewer**:

**Release note**:
```release-note
Fix issue with handler not releasing UnavailableNodeCount when killed during reconcile
```

### CODE REVIEW

File: controllers/nodenetworkconfigurationpolicy_controller.go [hunk 1]
```diff
@@ -303,6 +309,34 @@ func (r *NodeNetworkConfigurationPolicyReconciler) deleteEnactmentForPolicy(poli
...
+			if err != nil {
+				return err
+			}
+			instance.Status.UnavailableNodeCount = enactmentCountByCondition.Progressing()
+			return r.Client.Status().Update(context.TODO(), instance)
+		})
+	}
+	return nil
+}
+
```
Comment (qinqon) [inline controllers/nodenetworkconfigurationpolicy_controller.go:339]: I think we can wrap all on RetryOnConflict

```suggestion
func (r *NodeNetworkConfigurationPolicyReconciler) syncPolicyMaxUnavailableCount(policy *nmstatev1beta1.NodeNetworkConfigurationPolicy, ) error {
       return retry.RetryOnConflict(retry.DefaultRetry, func() error {
	    policyKey := types.NamespacedName{Name: policy.GetName(), Namespace: policy.GetNamespace()}
	    instance := &nmstatev1beta1.NodeNetworkConfigurationPolicy{}
	    err := r.Client.Get(context.TODO(), policyKey, instance)
	    if err != nil {
		    return err
	    }
	    _, enactmentCountByCondition, err := enactment.CountByPolicy(r.Client, instance)
	    if err != nil {
                return err
	    }
	    if instance.Status.UnavailableNodeCount != enactmentCountByCondition.Progressing() {
	        instance.Status.UnavailableNodeCount = enactmentCountByCondition.Progressing()
		return r.Client.Status().Update(context.TODO(), instance)
	    })
	}
	return nil
}

```

**Review by qinqon — CHANGES_REQUESTED** [review]

File: controllers/nodenetworkconfigurationpolicy_controller.go (hunk 1 above)
Comment (rhrazdil) [reply controllers/nodenetworkconfigurationpolicy_controller.go]: This looks better indeed. Updated

File: controllers/nodenetworkconfigurationpolicy_controller.go [hunk 2]
```diff
@@ -303,6 +309,26 @@ func (r *NodeNetworkConfigurationPolicyReconciler) deleteEnactmentForPolicy(poli
 	return nil
 }
 
+func (r *NodeNetworkConfigurationPolicyReconciler) syncPolicyMaxUnavailableCount(policy *nmstatev1beta1.NodeNetworkConfigurationPolicy) error {
```
Comment (qinqon) [inline controllers/nodenetworkconfigurationpolicy_controller.go:312]: let's call this `reconcilePolicyMaxUnavailableCount` the reconcile concept is well adopted at the context.

Comment (rhrazdil) [reply controllers/nodenetworkconfigurationpolicy_controller.go]: I see, thanks! [Done](https://github.com/nmstate/kubernetes-nmstate/compare/88b690394cbb1712666d3f42e9cf05b6331aa261..25193a39d16705ec55350aee2c9dd0f48c4c4d32)

**Review by qinqon — APPROVED** [review]

**Pull request merged**